File: pset6/cash.py
from cs50 import get_float

total_coins = 0

# Locate the index to perform integer division from the coins list
coins = [1, 5, 10, 25]

while True:
    cash = get_float("Change owed: ")
    if cash > 0:
        break

cents = cash * 100

# Reiterate backwards the index of coins
idx = len(coins) - 1

while idx >= 0:
    # perform integer division if the cents are bigger than the coins
    if coins[idx] <= cents:
        number_of_coins = int(cents/coins[idx])
        total_coins += number_of_coins
        cents -= number_of_coins * coins[idx]
    idx -= 1

print(total_coins)

File: pset1/cash.c
# include <stdio.h>
# include <cs50.h>
# include <math.h>

int coins;

int main(void)
{
    float dollars = get_float("Change owed: ");
    while (true)
    {
        if (dollars >= 0)
        {
            break;
        }
        else
        {
            dollars = get_float("Change owed: ");
        }
    }

    int cents = round(dollars * 100);
    do
    {
        // if can use quarters
        if (cents >= 25)
        {
            cents -= 25;
            coins++;
        }
        // if can use dimes
        else if (cents >= 10)
        {
            cents -= 10;
            coins++;
        }
        // if can use 5c
        else if (cents >= 5)
        {
            cents -= 5;
            coins++;
        }
        // if can use 1c
        else if (cents >= 1)
        {
            cents -= 1;
            coins++;
        }
    }
    while (cents > 0);
    printf("%i\n", coins);
}
